atomic_reactor/plugins/exit_koji_import.py

```diff
@@ -450,6 +455,23 @@ def upload_file(self, session, output, serverdir):
         self.log.debug("uploaded %r", path)
         return path
 
+    def upload_scratch_metadata(self, koji_metadata, koji_upload_dir, koji_session):
+        metadata_file = NamedTemporaryFile(prefix="metadata", suffix=".json", mode='wb')
+        metadata_file.write(json.dumps(koji_metadata).encode('utf-8'))
+        metadata_file.flush()
+
+        filename = "metadata.json"
+        meta_output = Output(file=metadata_file, metadata={'filename': filename})
+
+        try:
+            self.upload_file(koji_session, meta_output, koji_upload_dir)
+            path = os.path.join(koji_upload_dir, filename)
+            self.log = logging.LoggerAdapter(self.log, {'arch': METADATA_TAG})
```

Comment:
This is weird IMHO, with adapter there is no need to assign it to `self.log` and restoring it in finally:
```python
log = logging.LoggerAdapter(self.log, {'arch': METADATA_TAG})
log.info(path)
```

Reply:
How is actually that adapter supposed working here?

Reply:
Found it, `ArchFormatter` uses that

Reply:
yeah we are setting different arch tag which is used by ArchFormatter, to identify metadata log line

Reply:
changed